### PR TITLE
fix: OpenAIEmbeddings support for Azure OpenAI Service custom endpoints

### PR DESCRIPTION
**Pull request by zioproto**
Related issue #1560

### CODE REVIEW

**Comment by zioproto** [comment]
According to the docs:
https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#embeddings

The `POST` request should look like:
```
POST https://{your-resource-name}.openai.azure.com/openai/deployments/{deployment-id}/embeddings?api-version={api-version}
```

But I am setting `api_base` to `xxxxxx.openai.azure.com/openai` and I am getting:
```
https://xxxxxx.openai.azure.com/openai/engines/text-embedding-ada-002/embeddings
```

I am not sure where `engines` comes from

**Comment by zioproto** [comment]
I refactored the commit into 8179f728d3073765573a9c4ff07bb3ff717cb372

Now it is working for me as expected.

**Comment by zioproto** [comment]
I pushed a new commit to fix the linting

**Review by hwchase17 — APPROVED** [review]

File: langchain/embeddings/openai.py [hunk 1]
```diff
@@ -104,6 +106,13 @@ class OpenAIEmbeddings(BaseModel, Embeddings):
     client: Any  #: :meta private:
     model: str = "text-embedding-ada-002"
     deployment: str = model  # to support Azure OpenAI Service custom deployment names
+    openai_api_version: str = "2022-12-01"
+    openai_api_base: Optional[
+        str
+    ] = None  # to support Azure OpenAI Service custom endpoints
```
Comment (hwchase17) [inline langchain/embeddings/openai.py:112]:
```suggestion
    # to support Azure OpenAI Service custom endpoints
    openai_api_base: Optional[str] = None  
```

File: langchain/embeddings/openai.py [hunk 2]
```diff
@@ -104,6 +106,13 @@ class OpenAIEmbeddings(BaseModel, Embeddings):
...
+    openai_api_type: Optional[
+        str
+    ] = None  # to support Azure OpenAI Service custom endpoints
```
Comment (hwchase17) [inline langchain/embeddings/openai.py:115]:
```suggestion
    # to support Azure OpenAI Service custom endpoints
    openai_api_type: Optional[str] = None  
```

**Comment by aakashrshah** [comment]
Related issue: https://github.com/hwchase17/langchain/issues/3992

**Pull request merged**